### ISSUE TITLE
Type Classes and Row Types

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#1621]
We need to figure out how type classes and row types fit together.

For example, we can start with the question of whether/how we can implement `ToString` for a record. Even just for a fixed, closed record like `{ x = 123, y = 456}`.

**Comment by mlutze** [comment]
I don't think we should allow declaration of instances on structural types. Since they are structural, they could be used for different purposes in different parts of a program, so instances could end up conflicting.

Instead, we can create "magic instances" for derivable classes, which cover all record/schema types.

**Comment by magnus-madsen** [comment]
> 
> 
> I don't think we should allow declaration of instances on structural types. Since they are structural, they could be used for different purposes in different parts of a program, so instances could end up conflicting.
> 
> Instead, we can create "magic instances" for derivable classes, which cover all record/schema types.

Agreed.
